tests/test_futures.py

```diff
@@ -29,34 +29,6 @@ def test_nonprompt():
 
     assert(exists('analysis/topEFT/histos/output_check_yields_nonprompt.pkl.gz'))
 
-def test_make_yields():
-    args = [
-        "python",
-        "analysis/topEFT/get_yield_json.py",
-        "-f",
-        "analysis/topEFT/histos/output_check_yields.pkl.gz",
-        "-n",
-        "analysis/topEFT/output_check_yields"
-    ]
-
-    # Produce json
-    subprocess.run(args)
-
-def test_compare_yields():
-    args = [
-        "python",
-        "analysis/topEFT/comp_yields.py",
-        "analysis/topEFT/output_check_yields.json",
-        "analysis/topEFT/test/UL17_private_ttH_for_CI_yields.json",
-        "-t1",
-        "New yields",
-        "-t2",
-        "Ref yields"
-    ]
-
-    # Run comparison
-    subprocess.run(args)
-
```

Comment:
Maybe we should keep these after all, since the code coverage goes down a bit without them. I agree it's a bit redundant, but the prior test should halt the CI if things fail.

Reply:
Is it just that the code written to compare the yields (in other words the code that used only in this test and nowhere else) isn't covered?  That's not a good reason to keep it in the tests.  Instead you should remove the unused code from the repo.

Reply:
That's a fair point. I think some of it might be used elsewhere, but mostly for extracting yields.

Reply:
@klannon and @bryates, sorry if I'm just misunderstanding your points, but I think this code is definitely used, and definitely not something we want to get rid of. The code in question consists of the two scripts that are responsible for taking a pkl and getting the yields in JSON format, and for comparing the yields in two JSONs. These scripts are explicitly used in a later step of the CI to test check the output pkl file produced by the `test_futures` job. 

Previously, these scripts had been getting run in the `test_futures` as well. However, I'm not sure this was very useful, as the job was not even set up to fail if the yields disagreed, plus we have a separated, dedicated job for creating and comparing the yields anyway, so it seemed redundant (and potentially confusion-prone) to have it in the futures test as well. For these reasons, I just removed the yields checks from test_futures. 

However, I am not sure why that made the coverage go down, except for the fact that the dedicated jobs that create and compare the yields are just run directly in as jobs in the yaml file ([here](https://github.com/TopEFT/topcoffea/blob/python39/.github/workflows/main.yml#L66-L74)) i.e. not through `pytest`, so maybe they're not being tracked by codecov.

Reply:
> except for the fact that the dedicated jobs that create and compare the yields are just run directly in as jobs in the yaml file ([here](https://github.com/TopEFT/topcoffea/blob/python39/.github/workflows/main.yml#L66-L74)) i.e. not through `pytest`

That would explain it.

Reply:
Ok, so now that we understand why the codecov changed, @bryates I'm wondering if you still object to the removal of the yields test from the `test_futures` job?

Reply:
Ok, but I guess I'm still not completely sure hat I understand why we need to check the yields here. We have dedicated jobs for that later in the CI. To me, having the "run the processor" check separate from the "check the yields" check makes more sense since that way it's easier to see what's wrong if something fails. The way it's set up now (after you added the yields checks back into the futures job), if anything is wrong with the processor or if something unexpected changed that affects the yields, this job will fail (and that will make it harder to figure out what went wrong, and we'll have to spend more time digging through the CI output trying to find where the actual error came from).

Reply:
Let me think a bit about how we can keep the code coverage for these files, while avoiding what you described.

Reply:
@kmohrman would you be opposed to replacing these lines
https://github.com/TopEFT/topcoffea/blob/0e5642fe781386acf632542bf81aa6086f5cf4d4/.github/workflows/main.yml#L66-L74
with the `pytest` versions? Of course making sure to point to the `json` files from the "run the processor" section.

Reply:
Sure, that sounds reasonable to me. Though, just to make sure I'm following, when you say "point to the json files from the "run the processor" section", do you mean pkl files? Or am I missing something?

Reply:
Yes, sorry, point it to the `pkl` file.

Reply:
Ok, sounds good to me. Though, just to check, did you actually push the new purest files for the test_compare_yields_after_processor and test_make_yields_after_processor checks? I might be missing them, but don't see them in the list of changed files. Also, in the yaml file, should it be something like `tests/test_make_yields_after_processor.py` instead of just `test_make_yields_after_processor`?

Reply:
Thanks for catching that! I just added the new file.

Reply:
For the names, `pytest` looks for everything in the `tests` folder, and any python file _or_ any function name defined in any of them is able to be called.

Reply:
I should clarify, the `-k` flag allows this, which I had forgotten to add.